test/unit/async.spec.ts

```diff
@@ -0,0 +1,50 @@
+import { Expect, Test, TestCase } from "alsatian";
+import * as util from "../src/util";
+
+import { TranspileError } from "../../src/TranspileError";
+
+export class AsyncTests {
+
+    @Test("async function declaration")
+    public asyncFunctionDeclaration(): void {
+        const testCode = `
+            async function a() { }
+        `;
+        Expect(util.transpileString(testCode)).toBe(`require("lualib_bundle\");\na = __TS__Async(function()\nend);`);
```

Comment:
This is basically the same as translation tests, which are discouraged ([guidelines](https://github.com/Perryvw/TypescriptToLua/blob/master/CONTRIBUTING.md#testing-guidelines)).

Reply:
Hey, yes you are right, I just check if async stays in the code. Since it gets ignored in the current master, same as the await-keyword.

Functional tests for async-await are hard to include currently, since the promise-library is not yet shipped within the code.
However, I will probably include some self-written promise.ts into the lualib soon.